Aggiornamento termine stateful-apps #3109 (#3232)
cncf/glossary: content/it/stateful-apps.md

---
title: Stateful Apps
status: Completed
category: Proprietà
tags: ["fondamentali", "Proprietà", "applicazione"]
---

Esistono diversi modi per salvare lo stato di un'applicazione _stateful_.
Il più semplice è mantenere lo stato in memoria e non persisterlo.
Ma il problema è che ad ogni riavvio dell'applicazione verrebbe perso.
Per evitare che ciò accada, lo stato viene persistito in locale (su disco) o in database.

> Quando parliamo di applicazioni _stateful_ (letteralmente "a pieno stato") e stateless (letteralmente "[senza stato](/it/stateless-apps/)"),

Reviewer: Qui mi rifaccio alle diverse sfumature di significato https://www.collinsdictionary.com/dictionary/english/fula.Per me la più corretta sembra essere "characterized by, having", e dunque mi pare più utile tradurlo con "con", anche per metterlo in contrapposizione con il "senza" 
```suggestion
Quando parliamo di applicazioni _stateful_ (letteralmente "con stato") e stateless (letteralmente "[senza stato](/it/stateless-apps/)"),
```

il concetto di stato si riferisce a tutti i dati da salvare affinché l'applicazione funzioni come previsto. Qualsiasi tipo di esperienza di acquisto online che ti ricorda a che punto sta il tuo carrello è, ad esempio, un'applicazione _stateful_.

> Oggi, la maggior parte delle applicazioni che usiamo sono almeno in parte stateful. Tuttavia, negli ambienti [cloud native](/it/cloud-native-apps/), le applicazioni stateful rappresentano una sfida, perché le applicazioni cloud native sono molto dinamiche. Le applicazioni possono essere scalate verso l'alto o verso il basso, riavviate e spostate, ma devono comunque essere in grado di accedere al loro stato.

Reviewer: Sempre per evitare il falso amico "scalare"...
```suggestion
Oggi, la maggior parte delle applicazioni che usiamo sono almeno in parte stateful. Tuttavia, negli ambienti [cloud native](/it/cloud-native-apps/), le applicazioni stateful rappresentano una sfida, perché le applicazioni cloud native sono molto dinamiche. Le applicazioni sono rese scalabili verso l'alto o verso il basso, possono essere riavviate e spostate, ma devono comunque essere in grado di accedere al loro stato.
```

> Pertanto, le applicazioni stateful necessitano di un qualche tipo di storage accessibile, come i database.

Reviewer: Non sono totalmente sicuro che "like databases" si riferisca a
1. uno use-case che esemplifica l'idea della necessità di una memoria accessibile, e in questo caso dsi voglia dire che le stateful apps, esattamente come i database, necessita di una memoria
2. un database è una memoria accessibile adatta all'uso di una stateful app

Mi pare più giusta la seconda
```suggestion
Pertanto, le applicazioni stateful necessitano di un qualche tipo di memoria accessibile, come ad esempio un database.
```
